src/fileListUnzip.ts

```diff
@@ -0,0 +1,24 @@
+import { fileListFromZip } from './fileListFromZip';
+
+interface FileList {
+  name: string;
+  webkitRelativePath: string;
+  lastModified: number;
+  size: any;
+  text: () => Promise<string>;
+  arrayBuffer: () => Promise<ArrayBuffer>;
+}
+export async function fileListUnzip(fileList: FileList[]) {
```

Comment:
Please use the same definition as in fileListFromPath and the type 'PartialFileList' should probably be in its own file to be reused in both methods.

So in this case it should not be an array of FileList but just one FileList.

function appendFiles(fileList: PartialFileList, currentDir: string) {